apps/sms/views/inbox/js/inbox.js

```diff
@@ -674,13 +606,16 @@ var InboxView = {
         this.startRendering();
       }
 
-      this.appendThread(thread);
-
-      // Dispatch visually-loaded when rendered threads could fill up the top of
-      // the visible area.
-      if (--firstPanelCount === 0) {
-        this.emit('visually-loaded');
-      }
+      this.appendThread(thread).then(() => {
+        // Dispatch visually-loaded when rendered threads could fill up the top
+        // of the visible area.
+        if (--firstPanelCount === 0) {
+          this.list.setModel(this.listData);
+          this.emit('visually-loaded');
+        } else if (this.listData.length % this.FIRST_PANEL_THREAD_COUNT === 0) {
+          this.list.setModel(this.listData);
```

Comment:
__question:__ does this ```if (this.listData.length % this.FIRST_PANEL_THREAD_COUNT === 0)``` change to render new items on page-by-page basis, not item-by-item?